scripts/openzfs-merge.sh

```diff
@@ -0,0 +1,231 @@
+#!/bin/bash
+#
+# This script tries to merge OpenZFS commits to ZoL.
+#
+# Instruction:
+# 
+# Repository setup must be similar with openzfs-tracking.sh
+# requirements.
+#
+# Repository setup for valid compilation check:
+#	sh autogen.sh
+#	./configure --enable-debug
+#
+# mandatory git settings:
+#	[merge]
+#   	renameLimit = 999999
+#	[user]
+#       email = mail@gmelikov.ru
+#       name = George Melikov
+#
+# Copyright (c) 2016 George Melikov. All rights reserved.
+#
+
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+
+REPOSITORY_PATH='.'
+TMP_FILE='/tmp/gittmpmessage.txt'
+
+# list with potential OpenZFS commits
+UNPORTED_COMMITS_FILE=$SCRIPTDIR'/hashes.txt'
+
+# Next files will generate automatically
+# 	list with commits git can't merge automatically
+EXCEPTIONS_GIT=$SCRIPTDIR'/exceptions.txt'
+#	list with commits which can't be compiled without errors
+EXCEPTIONS_COMPILE=$SCRIPTDIR'/uncompiled.txt'
+#	list with commits which has cstyle error
+EXCEPTIONS_CSTYLE=$SCRIPTDIR'/unsctyled.txt'
+#	list with merged
+EXCEPTIONS_MERGED=$SCRIPTDIR'/merged.txt'
+
+LGREEN='\033[1;32m'		#  ${LGREEN}
+NORMAL='\033[0m'		#  ${NORMAL} 
+COUNT_MERGED=0
+LIST_MERGED=
+COUNT_COMPILE=0
+LIST_COMPILE=
+COUNT_CSTYLE=0
+LIST_CSTYLE=
+
+usage() {
+cat << EOF
+USAGE:
+$0 [-h] [-d directory] [-i commits.txt] [-c commit hash] [-g commit hash]
+
+DESCRIPTION:
+	Auto merge OpenZFS commits to ZFS on Linux git
+	repositories.
+	Result - git branch with name 'autoport-oz#issue'
+
+OPTIONS:
+	-h		Show this message
+	-d directory	Git repo with openzfs and zfsonlinux remotes
+	-i commits.txt	File with OpenZFS commit hashes to merge
+	(one hash per row)
+	-i commit hash	Prepare branch and try to merge this commit by hash,
+	leaves branch for manual merge if not successfull
+	-g commit hash	Generate commit description for existing commit in
+	branch generated by -i parameter
+
+EOF
+}
+
+clean_unmerged() {
+	git cherry-pick --abort
+	git checkout master
```

Comment:
It may be worth documenting that this script should be used on a clone where the origin is a fork of ZFS on Linux that has a remote to OpenZFS.